src/main/java/org/candlepin/resource/ConsumerResource.java

```diff
@@ -2316,13 +2324,25 @@ public Response bind(
             // I hate double negatives, but if they have accepted all
             // terms, we want comeToTerms to be true.
             long subTermsStart = System.currentTimeMillis();
-
-            if (subAdapter.hasUnacceptedSubscriptionTerms(owner.getKey())) {
-                return Response.serverError().build();
+            ExceptionMessage message;
+            try {
+                if (subAdapter.hasUnacceptedSubscriptionTerms(owner.getKey())) {
+                    message = new ExceptionMessage(
+                        i18n.tr("You must first accept Red Hat''s Terms and conditions. Please visit {0}",
+                        "https://www.redhat.com/wapps/tnc/ackrequired?site=candlepin" +
+                        "&event=attachSubscription"));
+                    return Response.serverError().entity(message).build();
+                }
+            }
+            catch (SubscriptionServiceException e) {
```

Comment:
Does this need to be handled in a nested try-catch? On the surface it looks like we can avoid this nesting and rearranging of code via finally without it.